db/memtable_list.cc

```diff
@@ -281,12 +514,16 @@ bool MemTableList::IsFlushPending() const {
 }
 
 // Returns the memtables that need to be flushed.
-void MemTableList::PickMemtablesToFlush(autovector<MemTable*>* ret) {
+void MemTableList::PickMemtablesToFlush(const uint64_t* max_memtable_id,
+                                        autovector<MemTable*>* ret) {
   AutoThreadOperationStageUpdater stage_updater(
       ThreadStatus::STAGE_PICK_MEMTABLES_TO_FLUSH);
   const auto& memlist = current_->memlist_;
   for (auto it = memlist.rbegin(); it != memlist.rend(); ++it) {
     MemTable* m = *it;
+    if (max_memtable_id != nullptr && m->GetID() > *max_memtable_id) {
+      continue;
+    }
```

Comment:
If there are memtables newer than `*max_memtable_id` that were not chosen for flush, is it correct to reset `flush_requested_` at the end of this function? In that case, I believe the newer memtables, which may already be eligible for flush, won't be considered for flush until another memtable is written.

Reply:
We discussed offline. `HasFlushRequested() { return flush_requested_l }` is called only in unit tests, thus does not affect the correctness of current logic.